### ISSUE TITLE
How to do sample by using GptManager

### ISSUE THREAD

**Issue by ljayx** [NVIDIA/TensorRT-LLM#193]
Hi,
From the C++ implementation of GptManager, it seems no where sampling config could be set. 
Is it supported? How could I do?

**Comment by jdemouth-nvidia** [comment]
Hi @ljayx ,

It certainly deserves more explanations in the documentation. Sorry about that. 

In the GptManager, you define the callback that generates new requests. It is an instance of [GetInferenceRequestsCallback](https://github.com/NVIDIA/TensorRT-LLM/blob/11e14500f35dd781b535ba009c906f55ecfee3b5/cpp/include/tensorrt_llm/batch_manager/callbacks.h#L31) and it’s passed in the [constructor](https://github.com/NVIDIA/TensorRT-LLM/blob/11e14500f35dd781b535ba009c906f55ecfee3b5/cpp/include/tensorrt_llm/batch_manager/GptManager.h#L50) of the class. 

Each instance of the list returned by that callback has a map of tensors associated with it. As you can see [here](https://github.com/NVIDIA/TensorRT-LLM/blame/11e14500f35dd781b535ba009c906f55ecfee3b5/cpp/include/tensorrt_llm/batch_manager/inferenceRequest.h#L189). The first thing the batch manager does is to convert that request into an [LlmRequest](https://github.com/NVIDIA/TensorRT-LLM/blob/11e14500f35dd781b535ba009c906f55ecfee3b5/cpp/include/tensorrt_llm/batch_manager/llmRequest.h#L37C17-L37C17) using the different tensor names listed [here](https://github.com/NVIDIA/TensorRT-LLM/blob/11e14500f35dd781b535ba009c906f55ecfee3b5/cpp/include/tensorrt_llm/batch_manager/GptManager.h#L106). As you can see the LlmRequest has a [SamplingConfig](https://github.com/NVIDIA/TensorRT-LLM/blob/11e14500f35dd781b535ba009c906f55ecfee3b5/cpp/include/tensorrt_llm/batch_manager/llmRequest.h#L172) that will be populated from the tensors attached to the InferenceRequest. 

So, long story short, you must attach your tensors of parameters to the InferenceRequest when you return the list of such requests in your callback. 

I hope it helps. 

Thanks,
Julien

**Comment by ljayx** [comment]
It is a very detailed explanation, thank a lot!

**Comment by ljayx** [comment]
Closed
